apps/pushes/views/diff.py

```diff
@@ -172,104 +173,57 @@ def diffLines(self, path, action):
             p = getParser(path)
         except UserWarning:
             return None
-        if action == 'added':
-            a_entities = []
-            a_map = {}
-        else:
+        old_data = OrderedDict()
+        if action != 'added':
             realpath = (action == 'moved' and self.moved[path] or
                         action == 'copied' and self.copied[path] or
                         path)
             content = self.content(realpath, self.rev1)
             try:
-                p.readContents(content)
-                a_entities, a_map = p.parse()
+                self.parse(p, content, old_data)
             except:
                 # consider doing something like:
                 # logging.warn('Unable to parse %s', path, exc_info=True)
                 return None
 
-        if action == 'removed':
-            c_entities, c_map = [], {}
-        else:
+        new_data = OrderedDict()
+        if action != 'removed':
             content = self.content(path, self.rev2)
             try:
-                p.readContents(content)
-                c_entities, c_map = p.parse()
+                self.parse(p, content, new_data)
             except:
                 # consider doing something like:
                 # logging.warn('Unable to parse %s', path, exc_info=True)
                 return None
         ar = AddRemove()
```

Comment:
`diff`, `differ` or even `add_remove` would all be better names for this instance.